internal/rbd/controllerserver.go

```diff
@@ -91,6 +92,37 @@ func (cs *ControllerServer) validateVolumeReq(ctx context.Context, req *csi.Crea
 		return err
 	}
 
+	err = validateStriping(req.Parameters)
+	if err != nil {
+		return status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	return nil
+}
+
+func validateStriping(parameters map[string]string) error {
+	stripeUnit := parameters["stripeUnit"]
+	stripeCount := parameters["stripeCount"]
+	if stripeUnit != "" && stripeCount == "" {
+		return errors.New("stripeCount must be specified when stripeUnit is specified")
+	}
+
+	if stripeUnit == "" && stripeCount != "" {
+		return errors.New("stripeUnit must be specified when stripeCount is specified")
+	}
+
+	objectSize := parameters["objectSize"]
+	if objectSize != "" {
+		objSize, err := strconv.ParseUint(objectSize, 10, 64)
+		if err != nil {
+			return fmt.Errorf("failed to parse objectSize %s: %w", objectSize, err)
+		}
+		// check objectSize is power of 2
+		if objSize == 0 || (objSize&(objSize-1)) != 0 {
```

Comment:
I don't immediately see how this "power of two" check works. A comment explaining the calculation would be nice.

Reply:
added a comment.